Lib9c/Action/ClaimStakeReward.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Immutable;
+using Bencodex.Types;
+using Libplanet;
+using Libplanet.Action;
+using Nekoyume.Model.Item;
+using Nekoyume.Model.State;
+using Nekoyume.TableData;
+using static Lib9c.SerializeKeys;
+
+namespace Nekoyume.Action
+{
+    public class ClaimStakeReward : GameAction
+    {
+        internal Address AvatarAddress { get; private set; }
+
+        public ClaimStakeReward(Address avatarAddress)
+        {
+            AvatarAddress = avatarAddress;
+        }
+
+        public ClaimStakeReward() : base()
+        {
+        }
+
+        public override IAccountStateDelta Execute(IActionContext context)
+        {
+            var states = context.PreviousStates;
+            if (!states.TryGetStakeState(context.Signer, out StakeState stakeState))
+            {
+                throw new InvalidOperationException();
+            }
+
+            var stakeRegularRewardSheet = states.GetSheet<StakeRegularRewardSheet>();
+            var stakeAchievementRewardSheet = states.GetSheet<StakeAchievementRewardSheet>();
+
+            var currency = states.GetGoldCurrency();
+            var balance = states.GetBalance(context.Signer, currency);
+
+            if (!stakeState.IsClaimable(context.BlockIndex))
+            {
+                throw new RequiredBlockIndexException();
+            }
+
+            var avatarState = states.GetAvatarState(AvatarAddress);
```

Comment:
```suggestion
            var avatarState = states.GetAvatarStateV2(AvatarAddress);
```

Reply:
I applied this suggestion with [95d1364](https://github.com/planetarium/lib9c/pull/962/commits/95d1364a97ca3eaf4240ff8f8647fcc1bbd523cc).